pylint/extensions/kwargs.py

```diff
@@ -0,0 +1,93 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import BoundMethod, UnboundMethod, nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import safe_infer
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class KeywordChecker(BaseChecker):
+    """Checks related to keyword arguments."""
+
+    name = "kwargs"
+    msgs = {
+        "W3501": (
+            "Call to `%s` missing keyword argument `%s`.",
+            "consider-using-keyword-argument",
+            "When using a literal directly in a function call, it's very hard to know what the argument is"
+            " if a positional argument is used because there's no variable name or attribute name to "
+            "rely on when reading. Using a keyword argument there's at least the attribute name to "
+            "help the reader understanding the call.",
+        ),
+    }
+
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check that called functions/methods are provided keyword arguments."""
+        called = safe_infer(node.func)
+
+        if not called:
+            return
+
+        needed_keywords, default_kwarg_names = self._get_args(called)
+
+        if len(needed_keywords) - len(default_kwarg_names) < 2:
+            # This checker will not apply if there are less than 2 positional args.
+            return
+
+        provided_kwarg_names = [
+            kwarg.arg for kwarg in node.keywords
+        ] + default_kwarg_names
+
+        for arg_name in needed_keywords:
+            if arg_name not in provided_kwarg_names:
+                self.add_message(
+                    "consider-using-keyword-argument",
+                    node=node,
+                    args=(
+                        called.name,
+                        arg_name,
+                    ),
+                    confidence=INFERENCE,
+                )
+
+    def _get_args(
+        self, node: nodes.ClassDef | nodes.FunctionDef | BoundMethod | UnboundMethod
+    ) -> tuple[list[str], list[str]]:
+        default_kwargs = []
+        needed_keywords = []
+
+        if isinstance(node, nodes.ClassDef):
+            needed_keywords = node.instance_attrs.keys()
+            if "__init__" in node.locals:
+                init_func = node.locals["__init__"][0]
+                if isinstance(init_func, nodes.FunctionDef):
```

Comment:
Can't you just use `self._get_args` again here?

Reply:
I don't understand your comment since this is within self._get_args

Reply:
Yes but you are duplicating logic of a few lines below. Why don't you call the same function recursively?

Reply:
I'm sorry but I don't think your comment applies. The code is slightly different for each if statement for a reason.